mode/verilog/verilog.js

```diff
@@ -354,11 +372,185 @@ CodeMirror.defineMode("verilog", function(config, parserConfig) {
   };
 });
 
-CodeMirror.defineMIME("text/x-verilog", {
-  name: "verilog"
-});
-CodeMirror.defineMIME("text/x-systemverilog", {
-  name: "systemverilog"
+  CodeMirror.defineMIME("text/x-verilog", {
+    name: "verilog"
+  });
+
+  CodeMirror.defineMIME("text/x-systemverilog", {
+    name: "verilog"
+  });
+
+  CodeMirror.defineMIME("text/x-svx", {
+    name: "verilog",
+    vxModeIsOn: true,
+    vxHooks: {
+        "\\": function(stream, state, thisMode) {
+               var codeRegion = "", vxIndent = 0, style = false;
+               var curPunc  = stream.string;
+               if ((0 == stream.pos) && (/\\SV/.test(stream.string))) {
+                 curPunc = (/\\SVX_version/.test(stream.string))
+                   ? "\\SVX_version" : stream.string;
+                 stream.skipToEnd();
+                 if (curPunc=="\\SV" && state.vxCodeActive) {codeRegion = "2v";};
+                 if (/\\SVX/.test(curPunc) && !state.vxCodeActive) {codeRegion ="2vx";};
+                 if (curPunc=="\\SVX_version" && state.vxCodeActive) {codeRegion = "2vx";};
+                 style=(thisMode.svxkeywords.propertyIsEnumerable(curPunc))
+                   ? thisMode.svxkeywords[curPunc] : "builtin";
+                 thisMode.svxCurCtlFlowChar  = thisMode.svxPrevPrevCtlFlowChar
```

Comment:
There are various points where you are assigning to properties of `thisMode`, which is, unless I'm really confused, referring to the `parserConfig` object passed to the mode. This object is shared by all instances of the mode (it is the value passed to `defineMIME`), and thus you probably shouldn't be changing it.